pkg/kubelet/apis/podresources/server_v1.go

```diff
@@ -27,22 +27,24 @@ import (
 	"k8s.io/kubelet/pkg/apis/podresources/v1"
 )
 
-// podResourcesServerV1alpha1 implements PodResourcesListerServer
+// podResourcesServerV1 implements PodResourcesListerServer
 type v1PodResourcesServer struct {
 	podsProvider    PodsProvider
 	devicesProvider DevicesProvider
 	cpusProvider    CPUsProvider
 	memoryProvider  MemoryProvider
+	draProvider     DRAProvider
 }
 
 // NewV1PodResourcesServer returns a PodResourcesListerServer which lists pods provided by the PodsProvider
 // with device information provided by the DevicesProvider
-func NewV1PodResourcesServer(podsProvider PodsProvider, devicesProvider DevicesProvider, cpusProvider CPUsProvider, memoryProvider MemoryProvider) v1.PodResourcesListerServer {
+func NewV1PodResourcesServer(podsProvider PodsProvider, devicesProvider DevicesProvider, cpusProvider CPUsProvider, memoryProvider MemoryProvider, draProvider DRAProvider) v1.PodResourcesListerServer {
 	return &v1PodResourcesServer{
 		podsProvider:    podsProvider,
 		devicesProvider: devicesProvider,
 		cpusProvider:    cpusProvider,
 		memoryProvider:  memoryProvider,
+		draProvider:     draProvider,
```

Comment:
Can this just embed the  new `PodResourcesProviders`, i.e.:
```
return &v1PodResourcesServer{
    providers PodResourcesProviders
}
```

Reply:
Is there a reason this direction wasn't taken in the latest iteration?

Reply:
yes, In the pod resource api we have v1[1]  and v1alpha1 [2] structs which hold the providers relevant for the version. 
So I don't think we should keep v1 and v1alpha1  as the same struct in this case. Maybe we can do it one v1alpha1 will be removed from the code. 

[1] -https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/podresources/server_v1.go#L31-L36
[2] - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/podresources/server_v1alpha1.go#L29-L32

Reply:
OK. thanks for clarifying.